src/storage/src/source/persist_source.rs

```diff
@@ -182,11 +177,11 @@ where
                             let session_cap = cap_set.delayed(&current_ts);
                             let mut session = output.session(&session_cap);
 
+                            // Give the batch to a random worker.
+                            let worker_idx = usize::cast_from(Instant::now().hashed()) % peers;
```

Comment:
we'd have to seed the RNG with something non-fixed (like the time) or we'd end up in with exactly the same problem we're fixing here. we could do that, but then (for example) it would be too spammy to print out the seed we used to logs

if, given the above, you still prefer the RNG, i'm happy to make the change :)